Code/Framework/AzCore/AzCore/Component/ComponentApplication.h

```diff
@@ -402,15 +395,17 @@ namespace AZ
         // from the m_console member when it goes out of scope
         AZ::SettingsRegistryConsoleUtils::ConsoleFunctorHandle m_settingsRegistryConsoleFunctors;
 
+#if !defined(_RELEASE)
```

Comment:
I know you only added this in response to other feedback, but unless the mere existence of this adds significant overhead (be that memory or performance) I think it's better for it to exist in all configurations and just have it do nothing in release (which I think is already the case because the macros check whether we're running in release or not).

Reply:
@AMZN-koppersr any opinions here? I personally don't feel strongly enough about it one way or another (rolling out the defines/declares across the engine and having a stable tracking location was the main goal of the PR).

Reply:
I don't mind terribly if we want to leave it as is, I just get wary whenever I see member variables conditionally compiled based on the configuration or platform:)

Reply:
Sounds good, should some (potentially immediate) use case for budget tracking arise (I should think memory in particular would be useful), we can roll this back in short order.

Reply:
With the AZ::Interface approach to gathering metrics it's not going to be a zero cost even if there's nothing in the AZ::Interface. It also means that users that don't compile the came leave the door open for somebody to hook up that interface and get extra information out of the application. So by default I think these doors should be closed. (Same goes for things like logging.) If the macros end up never calling this interface that's fine as well as that has the same net effect, but if the macros already have that `#if defined` might as well be extra sure and remove it here as well.

Reply:
I'll leave this as is and we can revisit it as a group as the interface and implementation evolves.